### ISSUE TITLE
What dropout layers to replace?

### ISSUE THREAD

**Issue by pashok3d** [bloodwass/mixout#4]
Hello,
Thank you for this interesting paper along with the code example.
As far as I understand you replace every dropout layer with mixout in BERT.
Did you experiment with replacing only specific dropout layers for BERT? 
Or on the other hand, what if we replace all dense layers, not only those followed by dropout layer?

**Comment by bloodwass** [comment]
In our paper, our main goal was the comparison of dropout and mixout. So we mainly presented the experiments replacing dropout with mixout. I personally applied mixout to all linear layers (not only those followed by dropout layers). In this experiment, mixout also significantly stabilizes fine-tuning results of BERT on a small training dataset with BERTAdam. However, you should know about recent studies such as [Revisiting Few-sample BERT Fine-tuning](https://arxiv.org/abs/2006.05987). This study pointed out that the main reason for the instability of BERT fine-tuning is BERTAdam. I strongly recommend this paper.
